File: bridge/workflows.js
import { readFileSync } from 'fs';
import { dirname, join } from 'path';
import { fileURLToPath } from 'url';

const here = dirname(fileURLToPath(import.meta.url));
const installedWorkflowRoot = join(here, 'cc_devtools', 'skills', 'frontend-devtools-workflows');
const repoWorkflowRoot = join(here, '..', 'cc_devtools', 'skills', 'frontend-devtools-workflows');
const workflowFiles = {
  inspect: 'SKILL.md',
  debug: 'references/debugging.md',
  selector: 'references/selectors.md',
  qa: 'references/qa.md',
  'local-data-patch': 'references/local-data-patch.md'
};

export function getWorkflowPrompt(name) {
  const key = Object.prototype.hasOwnProperty.call(workflowFiles, name) ? name : 'inspect';
  try {
    return readFileSync(join(installedWorkflowRoot, workflowFiles[key]), 'utf8');
  } catch {
    return readFileSync(join(repoWorkflowRoot, workflowFiles[key]), 'utf8');
  }
}

File: cc_devtools/workflows.py
from functools import lru_cache
from pathlib import Path


DEFAULT_WORKFLOW = "inspect"
WORKFLOW_DIR = Path(__file__).parent / "skills" / "frontend-devtools-workflows"
WORKFLOW_FILES = {
    "inspect": "SKILL.md",
    "debug": "references/debugging.md",
    "selector": "references/selectors.md",
    "qa": "references/qa.md",
    "local-data-patch": "references/local-data-patch.md",
}


@lru_cache(maxsize=None)
def get_workflow_prompt(name):
    key = name if name in WORKFLOW_FILES else DEFAULT_WORKFLOW
    path = WORKFLOW_DIR / WORKFLOW_FILES[key]
    return path.read_text(encoding="utf-8")
